### PR TITLE
New plugin: lnurlp

### PR DESCRIPTION
**Pull request by Kvaciral**
A simple http server to retrieve pay requests (bolt11s at the moment) to be used in lnurls

Based on the request-invoice plugin: it made sense to create a new plugin because having only one endpoint makes things more simple and secure
This one needs description_hash, while the request-invoice endpoint doesn't which would complicate things

Co-authored-by: W. J. van der Laan

### CODE REVIEW

**Comment by erikarvstedt** [comment]
Would it be possible to make rate-limiting optional?

This would allow implementing rate-limiting in the hosting stack (e.g., in the reverse proxy server).
Advantages:
- More fine-grained rate-limiting settings
- Improved performance (not very relevant here)
- It would allow sharing the same rate-limiting bucket with other invoice creation services.
  We plan to use this plugin alongside btcpayserver on [nixbitcoin.org/donate](https://nixbitcoin.org/donate) with a shared
  bucket.

File: lnurlp/lnurlp.py [hunk 1]
```diff
@@ -106,4 +106,16 @@ def init(options, configuration, plugin):
...
+plugin.add_option(
+    "lnurlp-ratelimit-rate",
+    "2 per minute",
+    "Change flask's ratelimiter, default is 2 per minute"
+)
+
+plugin.add_option(
+    "lnurlp-ratelimit-switch",
+    "enable",
+    "Flask's ratelimiter usage, enabled by default"
```
Comment (vmedea) [inline lnurlp/lnurlp.py:118]: please add the available options to the description here (`disable` and `enable`)

**Review by erikarvstedt — COMMENTED** [review]
Maybe merge `lnurlp-ratelimit-rate` and `lnurlp-ratelimit-switch` into option `lnurlp-rate-limit` which accepts extra value `disable`.

File: lnurlp/lnurlp.py [hunk 2]
```diff
@@ -30,7 +30,8 @@
 jobs = {}
 
 
-@limiter.limit(lambda: plugin.ratelimitrate, exempt_when=lambda: plugin.ratelimitswitch=="disable")
+#@limiter.limit(lambda: plugin.ratelimitrate, exempt_when=lambda: plugin.ratelimitswitch==False)
```
Comment (erikarvstedt) [inline lnurlp/lnurlp.py:33]: Leftover comment.

File: lnurlp/README.md [hunk 3]
```diff
@@ -140,9 +134,15 @@ You can combine multiple rate limits by separating them with a delimiter of your
...
-- `--lnurlp-ratelimit-rate="2 per 2 minutes;3 per hour;42 per day"`
-- `--lnurlp-ratelimit-rate="100/day, 500/2months"`
+- `--lnurlp-ratelimit="1 per minute"`
+- `--lnurlp-ratelimit="10/hour"`
+- `--lnurlp-ratelimit="2 per 2 minutes;3 per hour;42 per day"`
+- `--lnurlp-ratelimit="100/day, 500/2months"`
 
 Default is "2 per minute".
+
+To disable the use of the rate limiter:
```
Comment (erikarvstedt) [inline lnurlp/README.md:144]:
```suggestion
To disable rate limiting:
```

File: lnurlp/lnurlp.py [hunk 4]
```diff
@@ -116,7 +116,7 @@ def init(options, configuration, plugin):
 plugin.add_option(
     "lnurlp-ratelimit",
     "2 per minute",
-    "Change flask's ratelimiter, default is 2 per minute"
+    "Change flask's rate limiter, default is 2 per minute, `disable` to disable"
```
Comment (erikarvstedt) [inline lnurlp/lnurlp.py:119]:
```suggestion
    "Set flask's rate limiter, default is `2 per minute`, `disable` to disable"
```

**Comment by erikarvstedt** [comment]
[Fixup](https://github.com/erikarvstedt/plugins/commits/lnurlp-ea).
(Fetch with `git fetch https://github.com/erikarvstedt/plugins lnurlp-ea:lnurlp-ea`)

**Comment by cdecker** [comment]
@Mergifyio rebase

**Comment by cdecker** [comment]
@Mergifyio rebase

**Comment by chrisguida** [comment]
Needs rebase

**Comment by chrisguida** [comment]
I propose we add this plugin instead https://github.com/elsirion/clnurl

**Comment by Kvaciral** [comment]
This is going nowhere :-) .. perhaps @chrisguida 's plugin will, good luck!!

**Comment by chrisguida** [comment]
@Kvaciral if you want to add some tests, the CI will check everything from now on.

But I'll start moving forward on clnurl since you recommended it :)

**Pull request closed without merging**